src/west/app/config.py

```diff
@@ -182,14 +196,23 @@ def check_config(self, option):
             self.die(f'invalid configuration option "{option}"; '
                      'expected "section.key" format')
 
+    def _read(self, name, configfile):
+        value = self.config.get(name, configfile=configfile)
+        if value is None:
+            self.dbg(f'{name} is unset')
+            raise CommandError(returncode=1)
+        return value
+
     def read(self, args):
         self.check_config(args.name)
-        value = self.config.get(args.name, configfile=args.configfile or ALL)
-        if value is not None:
-            self.inf(value)
-        else:
-            self.dbg(f'{args.name} is unset')
-            raise CommandError(returncode=1)
+        value = self._read(args.name, args.configfile or ALL)
+        self.inf(value)
+
+    def append(self, args):
+        self.check_config(args.name)
+        value = self._read(args.name, args.configfile or LOCAL) or ''
+        args.value = value + args.value
+        self.write(args)
```

Comment:
The tricky bit here, is that if the name doesn't exist, the value isn't created and nothing is printed. The user needs to check the return value

```shell
$ west config -a non.existing whatever
$ echo $?
1
```

Together with the scenario you describe, where a separator is needed for the particular option, maybe this needs something more.
Perhaps the `-a` argument could be passed a separator value?

Reply:
Sure, I get your point. But aren't those 2 separate issues?

1. Report the "unable to find value to append to" error to the user in a more visible way (I can add that very easily)
2. Appending without a separator may confuse users